src/CTA.Rules.Config/Utils.cs

```diff
@@ -267,5 +275,27 @@ public static void DownloadFilesToFolder(string s3Bucket, string targetFolder, L
                 }
             });
         }
+
+        public static IEnumerable<string> GetProjectPaths(string solutionPath)
+        {
+            if(solutionPath.Contains(".sln") && File.Exists(solutionPath))
+            {
+                string solutionDir = Directory.GetParent(solutionPath).FullName;
+                IEnumerable<string> projectPaths = null;
+                try
+                {
+                    SolutionFile solution = SolutionFile.Parse(solutionPath);
+                    projectPaths = solution.ProjectsInOrder.Where(p => AcceptedProjectTypes.Contains(p.ProjectType)).Select(p => p.AbsolutePath);
+                }
+                catch(Exception ex)
+                {
+                    LogHelper.LogError(ex, $"Error while parsing solution file {solutionPath} falling back to directory parsing.");
+                    projectPaths = Directory.EnumerateFiles(solutionDir, "*.csproj", SearchOption.AllDirectories);
+                }
+
+                return projectPaths;
+            }
+            return null;
```

Comment:
Can we return an empty collection instead of null? This method replaces a `Directory.EnumerateFiles()` call which returns an empty collection when nothing is found.

Reply:
absolutely agreed